zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -330,6 +353,13 @@ public String getZnodeGroupAclServerDedicatedDomain() {
     return znodeGroupAclServerDedicatedDomain;
   }
 
+  public boolean shouldEnforceDedicatedDomain() {
+    if (znodeGroupAclServerShouldEnforceDedicatedDomain == null) {
+      setZnodeGroupAclServerShouldEnforceDedicatedDomain(System.getProperty(ENFORCE_DEDICATED_DOMAIN, "true"));
+    }
+    return Boolean.parseBoolean(znodeGroupAclServerShouldEnforceDedicatedDomain);
```

Comment:
should you do with ignoreCase ?? just in case, user types TRUE vs/vs true

Reply:
Good catch, but [Boolean.parseBoolean()](https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html#parseBoolean-java.lang.String-) is case insensitive (has an implicit ignoreCase).